src/realm-settings/KeysProvidersTab.tsx

```diff
@@ -131,8 +134,37 @@ export const KeysTabInner = ({ components, refresh }: KeysTabInnerProps) => {
   };
 
   const onDragFinish = (itemOrder: string[]) => {
-    setItemOrder(["data", ...itemOrder.filter((i) => i !== "data")]);
+    setItemOrder(itemOrder);
     setLiveText(t("common:onDragFinish"));
+    const updateAll = components.map((component: ComponentData) => {
+      const componentToSave = { ...component };
+      componentToSave.providerDescription = undefined;
+
+      return adminClient.components.update(
+        { id: component.id! },
+        {
+          ...componentToSave,
+          parentId: component?.parentId,
+          providerId: component?.providerId,
+          providerType: KeyProviderType,
+          config: {
+            priority: [
+              (
+                itemOrder.length -
+                itemOrder.indexOf(component.id || "") +
```

Comment:
I think this is one of those cases where `id` is always present but the types are not correct. You can use a non-null assertion here to fix that.
```suggestion
                itemOrder.indexOf(component.id!) +
```